noir-projects/aztec-nr/aztec/src/keys/assert_public_key_freshness.nr

```diff
@@ -0,0 +1,67 @@
+use dep::protocol_types::{
+  address::{
+    AztecAddress,
+    PartialAddress
+  }, 
+  constants::{
+    GENERATOR_INDEX__PUBLIC_KEYS_HASH,
+    GENERATOR_INDEX__CONTRACT_ADDRESS_V1,
+    CANONICAL_KEY_REGISTRY_ADDRESS
+  },
+  grumpkin_point::GrumpkinPoint,
+};
+
+use crate::context::PrivateContext;
+use crate::hash::{
+  pedersen_hash, 
+  poseidon2_hash,
+};
+use crate::oracle;
+use crate::state_vars::shared_mutable::shared_mutable_private_getter::SharedMutablePrivateGetter;
+
+struct PublicKeyTypeEnum {
+  NULLIFIER: u8,
+}
+
+global PublicKeyType = PublicKeyTypeEnum {
+  NULLIFIER: 0,
+};
+
+pub fn assert_nullifier_public_key_is_fresh(
+  context: &mut PrivateContext,
+  address: AztecAddress, 
+  nullifier_public_key_to_test: GrumpkinPoint,
+) {
+  // This is the storage slot of the nullifier_public_key inside the key registry contract
+  let storage_slot_of_nullifier_public_key = 1;
```

Comment:
On this, could we not expose "global" values on the struct directly? Seems like it could be super sleek it we could do something like
```
registry::SOME_STORAGE_SLOT_CONSTANT
```

@Thunkar do you have any thoughts here? I have actually not checked if this is already possible with the code you made 👀